src/dbnode/storage/namespace.go

```diff
@@ -348,12 +349,48 @@ func newDatabaseNamespace(
 		metrics:                newDatabaseNamespaceMetrics(scope, iops.MetricsSamplingRate()),
 	}
 
+	n.reInitPools()
 	n.initShards(nopts.BootstrapEnabled())
 	go n.reportStatusLoop()
 
 	return n, nil
 }
 
+func (n *dbNamespace) reInitPools() {
+	schemaInjector := func(e encoding.SchemaInjectable) encoding.SchemaInjectable {
+		if schema, ok := n.SchemaHistory().GetLatest(); ok {
```

Comment:
is schema history synchronized?

Reply:
Oh I see the namespace locks it...not a huge fan of this because now random parts of the code will be acquiring the namespace lock. Would prefer if there was an independent lock for the registry if that makes sense

Reply:
> Oh I see the namespace locks it...not a huge fan of this because now random parts of the code will be acquiring the namespace lock. Would prefer if there was an independent lock for the registry if that makes sense

With database schema registry approach, you'd have to lock it twice, first time to lock db to access OwnedNamespaces, second time to lock dbNamespace to access schema history.

Reply:
I see by "registry" you mean "schema history". yes, we can have a lock just for schema history (instead of a namespace wide lock).